frappe/public/js/frappe/form/controls/color.js

```diff
@@ -48,7 +48,24 @@ frappe.ui.form.ControlColor = frappe.ui.form.ControlData.extend({
 			$(window).off('hashchange.color-popover');
 		});
 
-		this.$wrapper.find('.control-input').on('click', (e) => {
+		this.picker.on_change = (color) => {
+			this.set_value(color);
+		};
+
+		if (!this.selected_color) {
+			this.selected_color = $(`<div class="selected-color"></div>`);
+			this.selected_color.insertAfter(this.$input);
+		}
+
+		if (!this.$wrapper.find('.control-input').get(0)) {
+			this.$wrapper.find('.selected-color')
+				.css({
+					"top": "calc(23% + 1px)",
+					"z-index": "2"
+				});
+		}
```

Comment:
you can move the color picker related to style to `color_picker.scss`. Also, "23%" seems arbitrary, it may break with different grid sizes (verify once).

Reply:
Ok will check.